### PR TITLE
crud and updated server

### PR DESCRIPTION
**Pull request by jonpaul108**
nothing to compare. Forgot to commit on the correct branch! But please check my code

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -18,20 +19,65 @@ app.get('/api/buytest', (req, res) => {
...
       res.send(docs);
-    }    
+    }
   });
 });
 
-app.get('/api/bpowertest', (req, res) =>{
-  dpower.find((err, docs) => {
+app.get('/api/buytest/one/:id', (req, res) => {
+  const id = JSON.parse(req.params.id);
```
Comment (yukiej) [inline server/index.js]: Do you need JSON.parse here? (Have you tried it without?)

File: server/index.js [hunk 2]
```diff
@@ -18,20 +19,65 @@ app.get('/api/buytest', (req, res) => {
...
+  modalBuy.read(id, (err, doc) => {
     if (err) {
-      res.send(err);
+      res.status(500).send(err);
     } else {
-      res.send(docs);
-    }    
-  });
+      res.status(200).send(JSON.stringify(doc))
```
Comment (yukiej) [inline server/index.js]: You might be able to shorten this to res.json(doc).status(200)

File: server/index.js [hunk 3]
```diff
@@ -18,20 +19,65 @@ app.get('/api/buytest', (req, res) => {
...
 });
 
-app.get('/api/bpowertest', (req, res) =>{
-  dpower.find((err, docs) => {
+app.get('/api/buytest/one/:id', (req, res) => {
+  const id = JSON.parse(req.params.id);
+  modalBuy.read(id, (err, doc) => {
     if (err) {
-      res.send(err);
+      res.status(500).send(err);
```
Comment (yukiej) [inline server/index.js]: This is probably ok error handling, but you might also try looking into express's built-in error handling to see what is standard

File: server/index.js [hunk 4]
```diff
@@ -18,20 +19,65 @@ app.get('/api/buytest', (req, res) => {
...
+  const param = req.body;
+  modalBuy.create(param, (err, success) => {
+    if (err) {
+      res.status(500).send();
+    } else {
+      res.status(201).send('lovin life');
+    }
+  })
+});
+// app.post('/api/bpowertest', (req, res) => {
```
Comment (yukiej) [inline server/index.js]: You can probably delete this if you aren't using it

File: server/index.js (hunk 1 above)
Comment (yukiej) [reply server/index.js]: (same for lines 58 and 70)

File: server/index.js [hunk 5]
```diff
@@ -18,20 +19,65 @@ app.get('/api/buytest', (req, res) => {
...
-    }    
+    }
   });
 });
 
-app.get('/api/bpowertest', (req, res) =>{
-  dpower.find((err, docs) => {
+app.get('/api/buytest/one/:id', (req, res) => {
+  const id = JSON.parse(req.params.id);
+  modalBuy.read(id, (err, doc) => {
```
Comment (yukiej) [inline server/index.js]: Related to the conversation we had with Kent earlier: Not necessarily, but might be helpful for future work to factor out these Mongoose queries to a separate file for better separation of concerns and more database modularity (so when we swap out the database later, you don't have to change the code in this file as much).

File: server/index.js (hunk 3 above)
Comment (jonpaul108) [reply server/index.js]: Great. Taking a look.

File: server/index.js (hunk 4 above)
Comment (jonpaul108) [reply server/index.js]: Deleted!

File: server/index.js (hunk 5 above)
Comment (jonpaul108) [reply server/index.js]: Right. I like the idea. I think I'll use a routing page to connect to controllers.
